### DISCUSSION TITLE
`gitlabci-include` manager doesn't work with wildcards

### DISCUSSION THREAD

**Discussion by wolfgangwalther** [renovatebot/renovate#14398] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

31.89.5

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

14.7.3

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

With a `.gitlab-ci.yml` file like this:

```yaml
include: ".gitlab/jobs/*.yaml"
```

and some valid job files in that directory (CI is working), I get the following in the log:

```
DEBUG: Empty or non existent gitlabci file (repository=[...])
       "file": ".gitlab/jobs/*.yaml"
```

The wildcard is taken literally instead of being expanded to the job files.

[Wildcards are supported](https://docs.gitlab.com/ee/ci/yaml/includes.html#use-includelocal-with-wildcard-file-paths) by GitLab:

> You can use wildcard paths (`*` and `**`) with `include:local`. 

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
This isn't a bug. This isn't simply supported by renovate.

**Reply by wolfgangwalther** [reply]
Now, that I understand what `gitlabci-include` does... I still think it's a bug. I didn't confirm that, but if I understand correctly, then `gitlabci-include` will fail to update a external include that is inside a locally included file - when the local include has a wildcard.

So `gitlabci-include` would still have to traverse the local includes to actually get to all external includes...

This can be worked around by additional `fileMatch`es - but that's the same as with `gitlabci` itself. And that supports wildcards.

So... I still think there is room for improvement.

**Reply by viceice** [reply]
No, `gitlabci-include` manager will be merged to `gitlabci` manager soon. And supporting wildcards is a feature which isn't supported yet.

`fileMatch` is used on a very different layer of renovate, the manager itself only getting the results of that match.

**Reply by wolfgangwalther** [reply]
> No, gitlabci-include manager will be merged to gitlabci manager soon.

That's much better. And would solve the problem, **if** `gitlabci` were to support wildcards.

> And supporting wildcards is a feature which isn't supported yet.

Ah, ok. I was mislead by this test-case:

https://github.com/renovatebot/renovate/blob/a674f727dd489264cb8bd0bb524326802ab33c39/lib/manager/gitlabci/__fixtures__/gitlab-ci.3.yaml#L15

But there is no file matching that include.

**Comment by wolfgangwalther** [comment]
If you didn't consider this a bug, wouldn't this be a feature request then? Am I missing something fundamental why this wouldn't be possible to support?

**Comment by wolfgangwalther** [comment]
Ah, I think I understand now. `gitlabci-include` is about something different - basically everything **except** `local`.
